### PR TITLE
feat: major update to jmeter executor

### PR DESCRIPTION
**Pull request by dejanzele**
## Pull request description 

* Add support for sending artifacts to Cloud
* Add documentation on local dev and testing in Kubernetes
* Generate report by default

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: pkg/runner/runner.go [hunk 1]
```diff
@@ -68,16 +85,19 @@ func (r *JMeterRunner) Run(execution testkube.Execution) (result testkube.Execut
 
 	path := ""
 	workingDir := ""
+	basePath, _ := filepath.Abs(r.Params.DataDir)
```
Comment (vsukhin) [inline pkg/runner/runner.go]: log errror, just in case

File: pkg/runner/runner.go [hunk 2]
```diff
@@ -24,24 +32,33 @@ func NewRunner() (*JMeterRunner, error) {
 		return nil, fmt.Errorf("could not initialize JMeter runner variables: %w", err)
 	}
 
-	return &JMeterRunner{
+	r := &JMeterRunner{
```
Comment (exu) [inline pkg/runner/runner.go:30]: Can we close this in some library and don't leak cloud implementation details into the executor please.

Comment (exu) [reply pkg/runner/runner.go:30]: Maybe some convention will be enough here for all possible executors who needs grpc connection

File: pkg/runner/runner.go [hunk 3]
```diff
@@ -153,18 +187,58 @@ func (r *JMeterRunner) Run(execution testkube.Execution) (result testkube.Execut
...
+func (r *JMeterRunner) scrape(ctx context.Context, dirs []string, execution testkube.Execution) (err error) {
+	if !r.Params.ScrapperEnabled {
+		return nil
+	}
+
+	output.PrintLog(fmt.Sprintf("%s Extracting artifacts from %s using filesystem extractor", ui.IconCheckMark, dirs))
+	extractor := scraper.NewFilesystemExtractor(dirs, filesystem.NewOSFileSystem())
+	var loader scraper.Uploader
+	var meta map[string]any
+	if r.Params.CloudMode {
```
Comment (exu) [inline pkg/runner/runner.go]: again leaking implementation details

File: pkg/runner/runner.go [hunk 4]
```diff
@@ -153,18 +187,58 @@ func (r *JMeterRunner) Run(execution testkube.Execution) (result testkube.Execut
...
+			r.Params.Location,
+			r.Params.Token,
+			r.Params.Bucket,
+			r.Params.Ssl,
+		)
+		if err != nil {
+			return err
+		}
+	}
+	scraperV2 := scraper.NewScraperV2(extractor, loader)
```
Comment (exu) [inline pkg/runner/runner.go]: why scrapper V2 - we were discussing it some time ago

**Review by vsukhin — APPROVED** [review]
looks good, couple of minor notes

File: pkg/runner/runner.go [hunk 5]
```diff
@@ -153,18 +187,58 @@ func (r *JMeterRunner) Run(execution testkube.Execution) (result testkube.Execut
...
+		if err := r.scrape(context.Background(), directories, execution); err != nil {
 			return *executionResult.WithErrors(fmt.Errorf("scrape artifacts error: %w", err)), nil
 		}
 	}
 
 	return executionResult, nil
 }
 
+func (r *JMeterRunner) scrape(ctx context.Context, dirs []string, execution testkube.Execution) (err error) {
+	if !r.Params.ScrapperEnabled {
```
Comment (vsukhin) [inline pkg/runner/runner.go]: looks like this part can be shared among executors

File: pkg/runner/runner.go (hunk 2 above)
Comment (dejanzele) [reply pkg/runner/runner.go:30]: we can hide it a bit better than this, but cannot fully hide it from users

File: pkg/runner/runner.go (hunk 4 above)
Comment (dejanzele) [reply pkg/runner/runner.go]: yeah, old scraper is removed, will drop the V2

File: pkg/runner/runner.go (hunk 5 above)
Comment (dejanzele) [reply pkg/runner/runner.go]: updated

**Pull request merged**